aten/src/ATen/native/EmbeddingBag.cpp

```diff
@@ -95,6 +158,67 @@ static void index_select_scale_add(const Tensor &select_indices,
   }
 }
 
+template<>
+void index_select_scale_add<float>(const Tensor &select_indices,
+                                          const Tensor &add_indices,
+                                          const Tensor &scale,
+                                          const Tensor &src,
+                                          Tensor &output,
+                                          const Tensor& offsets) {
+  int64_t ddim = src.size(1);
+  auto* scale_data = scale.data<float>();
+  auto select_indices_data = select_indices.data<int64_t>();
```

Comment:
Do we actually assert that these data types match somewhere?

Reply:
@cpuhrsch i think this does it: https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/EmbeddingBag.cpp#L214

Also there's a PR out to fix that location: https://github.com/pytorch/pytorch/pull/19432